File: src/main/java/com/kingdom/model/computer/MediumComputerPlayer.java
package com.kingdom.model.computer;

import com.kingdom.model.Card;
import com.kingdom.model.Game;
import com.kingdom.model.Player;

public class MediumComputerPlayer extends ComputerPlayer {
    public MediumComputerPlayer(Player player, Game game) {
        super(player, game);
        difficulty = 2;
    }

    @Override
    protected void setupStartingStrategies() {
        if (hasGardens && hasExtraBuys) {
            gardensStrategy = true;
        } else if (kingdomCardMap.containsKey("Chapel")) {
            chapelStrategy = true;
        } else if (hasDuke) {
            dukeStrategy = true;
        }
    }

    @Override
    protected boolean excludeCard(Card card) {
        return excludeCardMedium(card);
    }
}

File: src/main/kotlin/com/kingdom/model/computer/MediumComputerPlayer.kt
package com.kingdom.model.computer

import com.kingdom.model.cards.Card
import com.kingdom.model.OldGame
import com.kingdom.model.OldPlayer

class MediumComputerPlayer(player: OldPlayer, game: OldGame) : ComputerPlayer(player, game) {
    init {
        difficulty = 2
    }

    override fun setupStartingStrategies() {
        if (hasGardens && hasExtraBuys) {
            isGardensStrategy = true
        } else if (kingdomCardMap.containsKey("Chapel")) {
            chapelStrategy = true
        } else if (hasDuke) {
            dukeStrategy = true
        }
    }

    override fun excludeCard(card: Card): Boolean {
        return excludeCardMedium(card)
    }
}
